CMakeLists.txt

```diff
@@ -0,0 +1,857 @@
+       OFF)
+if(gperftools_dynamic_sized_delete_support)
+  set(ENABLE_DYNAMIC_SIZED_DELETE 1)
+endif()
+
+option(gperftools_sized_delete "Build sized delete operator" OFF)
+if(gperftools_sized_delete)
+  set(ENABLE_SIZED_DELETE 1)
+endif()
+
+set(CMAKE_REQUIRED_FLAGS -fsized-deallocation)
+check_cxx_source_compiles("
+  #include <new>
+  int main() { (::operator delete)(0, 256); return 0; }"
+  have_sized_deallocation)
+set(CMAKE_REQUIRED_FLAGS)
+
+check_c_source_compiles(
+  "namespace abc { void foo() {} } int main() { abc::foo(); return 0; }"
+  HAVE_NAMESPACES)
+
+check_c_source_compiles("
+  #include <unwind.h>
+  int main()
+  {
+    &_Unwind_Backtrace;
+    return 0;
+  }"
+  HAVE_UNWIND_BACKTRACE)
+
+if(enable_backtrace)
+  set(default_emergency_malloc ON)
+else()
+  set(default_emergency_malloc OFF)
+endif()
+
+if(will_use_libunwind AND ARM)
+  set(default_emergency_malloc ON)
+endif()
+
+option(gperftools_emergency_malloc
+       "Build emergency malloc"
+       ${default_emergency_malloc})
+
+check_c_source_compiles(
+  "int main() { void* sp = __builtin_stack_pointer(); return 0; }"
+  HAVE_BUILTIN_STACK_POINTER)
+check_c_source_compiles(
+  "int main() { return __builtin_expect(main != 0, 1); }"
+  HAVE_BUILTIN_EXPECT)
+
+check_c_source_compiles("
+  #include <unistd.h>
+  int main()
+  {
+    char** env = __environ;
+    return 0;
+  }"
+  HAVE___ENVIRON)
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+   CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.1.2")
+  message(WARNING "gcc has this bug: http://gcc.gnu.org/ml/gcc-bugs/2006-09/msg02275.html")
+elseif(APPLE)
+  message(WARNING "OSX __thread support is known to call malloc which makes "
+                    "it unsafe to use from malloc replacement")
+elseif(MINGW)
+  message(WARNING "mingw doesn't really support tls")
+else()
+  check_c_source_compiles("static __thread int p = 0;" HAVE_TLS)
+endif()
+
+if(need_nanosleep)
+  check_c_source_compiles(
+    "#include <time.h>
+     int main()
+     { static struct timespec ts; nanosleep(&ts, NULL); return 0; }"
+    nanosleep_ok)
+  if(NOT nanosleep_ok)
+    set(CMAKE_REQUIRED_LIBRARIES rt)
+    check_c_source_compiles(
+      "#include <time.h>
+       int main()
+       { static struct timespec ts; nanosleep(&ts, NULL); return 0; }"
+      nanosleep_ok)
+    if(nanosleep_ok)
+      set(nanosleep_libs rt)
+    else()
+      message(FATAL_ERROR "cannot find the nanosleep function")
+    endif()
+    set(CMAKE_REQUIRED_LIBRARIES)
+  endif()
+endif()
+
+if(enable_backtrace)
+  check_symbol_exists("backtrace" "execinfo.h" HAVE_DECL_BACKTRACE)
+  check_function_exists("backtrace" backtrace_exists)
+  if(NOT backtrace_exists)
+    set(CMAKE_REQUIRED_LIBRARIES execinfo)
+    check_function_exists("backtrace" backtrace_exists)
+    set(CMAKE_REQUIRED_LIBRARIES)
+    if(backtrace_exists)
+      list(INSERT unwind_libs 0 execinfo)
+    endif()
+  endif()
+endif()
+
+find_package(Threads REQUIRED)
```

Comment:
Are threads _actually_ required? I recall being able to disable threading for tcalloc.

Reply:
Let me look into this.